core/unit_test/TestGraphAtomicLocks.hpp

```diff
@@ -0,0 +1,103 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Graph.hpp>
+
+#include <gtest/gtest.h>
+
+namespace Test {
+
+#if defined(KOKKOS_ENABLE_CUDA)
+#define BACKEND_SPACE_ATOMIC_LOCKS_DEVICE CUDA_SPACE_ATOMIC_LOCKS_DEVICE
+#elif defined(KOKKOS_ENABLE_HIP)
+#define BACKEND_SPACE_ATOMIC_LOCKS_DEVICE HIP_SPACE_ATOMIC_LOCKS_DEVICE
+#elif defined(KOKKOS_ENABLE_SYCL)
+#define BACKEND_SPACE_ATOMIC_LOCKS_DEVICE SYCL_SPACE_ATOMIC_LOCKS_DEVICE
+#else
+#error "This test can only be run when either Cuda or HIP is enabled."
+#endif
+
+struct CheckAtomicLocksOnDeviceFunctor {
+  template <typename T>
+  KOKKOS_FUNCTION void operator()(const T,
+                                  [[maybe_unused]] bool& cur_result) const {
+    KOKKOS_IF_ON_DEVICE(
+        cur_result =
+            cur_result &&
+            (desul::Impl::BACKEND_SPACE_ATOMIC_LOCKS_DEVICE != nullptr);)
+  }
+};
```

Comment:
I think this test here is somewhat problematic. It relies on desul implementation details, not something I am keen on seeing in Kokkos. I think the atomic_add test is good enough.

Reply:
OK. I've deleted this test that relies on the desul implementation details. Note that I've also changed a bit the atomic add test. The previous version of the atomic add test wasn't OK because `Kokkos::complex` is not a trivial type, hence the initialisation triggered a kernel launch. So I now initialise the data view as part of the graph.